### PR TITLE
feat: add dockerfile, service.yml and deployment.yml files to auth se…

### PR DESCRIPTION
**Pull request by scott45**
# SPDX-License-Identifier: Apache-2.0

## What did we change?
Add a dockerfile, service.yml and deployment.yml to auth-service.

## Why are we doing this?
They are required as deployable files

## How was it tested?
- [x] Locally
- [x] Development Environment
- [ ] Not needed, changes very basic
- [ ] Husky successfully run
- [ ] Unit tests passing and Documentation done

Screenshot 2024-08-01 at 11 06 23 (2)

### CODE REVIEW

**Comment by cshezi** [comment]
Let's answer PR questioners

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ENV NODE_ENV=production
+ENV MAX_CPU=
+
+#Apm
+ENV APM_ACTIVE=true
+ENV APM_URL=http://apm-server.development.svc.cluster.local:8200/
+ENV APM_SECRET_TOKEN=
+
+# Database
+ENV DATABASE_URL=
```
Comment (cshezi) [inline Dockerfile:54]: Environment variables are not being used in this service e.g REDIS

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# Redis
+ENV REDIS_DB=0
+ENV REDIS_AUTH=
+ENV REDIS_SERVERS=
+ENV REDIS_IS_CLUSTER=
+ENV CACHETTL=300
+
+# Set healthcheck command
+HEALTHCHECK --interval=60s CMD [ -e /tmp/.lock ] || exit 1
+EXPOSE 4222
```
Comment (cshezi) [inline Dockerfile:69]: Consider exposing the rest port this port is used for NATS subscription

**Review by cshezi — CHANGES_REQUESTED** [review]

**Review by cshezi — CHANGES_REQUESTED** [review]
We are missing a couple of Environment variables

image

**Comment by cshezi** [comment]
Please try starting up the image and try GET`/health` endpoint to see if the service is up and running

**Review by cshezi — APPROVED** [review]
LGTM

**Review by Justus-at-Tazama — APPROVED** [review]

**Pull request merged**